app/src/server/tasks/evaluateTestSetEntries.task.ts

```diff
@@ -0,0 +1,428 @@
+      (firstResult.status !== "PENDING" &&
+        firstResult.status !== "ERROR" &&
+        secondResult.status !== "PENDING" &&
+        secondResult.status !== "ERROR")
+    )
+      return;
+
+    await prisma.datasetEvalResult.updateMany({
+      where: {
+        id: {
+          in: [firstResult.id, secondResult.id],
+        },
+      },
+      data: {
+        status: "IN_PROGRESS",
+        errorMessage: null,
+      },
+    });
+
+    const entries = [];
+    const entryIds = [];
+
+    for (const result of [firstResult, secondResult]) {
+      try {
+        let entry;
+        if (result.datasetEvalOutputSource.modelId !== ORIGINAL_MODEL_ID) {
+          entry = await prisma.fineTuneTestingEntry.findUniqueOrThrow({
+            where: {
+              modelId_datasetEntryId: {
+                modelId: result.datasetEvalOutputSource.modelId,
+                datasetEntryId: result.datasetEvalDatasetEntry.datasetEntryId,
+              },
+            },
+            include: {
+              fineTune: true,
+            },
+          });
+          if (isComparisonModel(entry.modelId)) {
+            entryIds.push(getComparisonModelName(entry.modelId));
+          } else if (entry.fineTune) {
+            entryIds.push("openpipe:" + entry.fineTune.slug);
+          } else {
+            throw new Error("No fineTune or comparison model found for entry");
+          }
+        } else {
+          entry = result.datasetEvalDatasetEntry.datasetEntry;
+          entryIds.push("the original model");
+        }
+        entries.push(entry);
+      } catch (e) {
+        console.error("error getting entry for result", result, e);
+        await prisma.datasetEvalResult.updateMany({
+          where: {
+            id: {
+              in: [firstResult.id, secondResult.id],
+            },
+          },
+          data: {
+            status: "ERROR",
+            errorMessage: "Error retrieving relevant input for evaluation",
+          },
+        });
+        return;
+      }
+    }
+
+    const [firstEntry, secondEntry] = entries;
+    const [firstEntryId, secondEntryId] = entryIds;
+
+    if (!firstEntry?.output || !secondEntry?.output || !firstEntryId || !secondEntryId) {
+      await prisma.datasetEvalResult.updateMany({
+        where: {
+          id: {
+            in: [firstResult.id, secondResult.id],
+          },
+        },
+        data: {
+          status: "ERROR",
+          errorMessage: "Error preparing for evaluation",
+        },
+      });
+      return;
+    }
+
+    let explanation;
+    let judgement;
+
+    const instructions = firstResult.datasetEvalDatasetEntry.datasetEval.instructions;
+
+    try {
+      const input: ChatCompletionCreateParams = {
+        model: "gpt-4",
+        messages: [
+          {
+            role: "system",
+            content: `You are an intelligent and fair judge of chatbots. Evaluate the following two chatbot responses and choose which one is better in the following way: ${
```

Comment:
You are an intelligent and fair judge of chatbots. Evaluate the following two chatbot responses and choose which one is better. If the outputs are of similar quality, you can mark them as `EQUAL`.

And then, if there are instructions, add the following:

The user has provided the following instructions on what you should evaluate the outputs on: <instructions>